src/lib/http/proxy-agent.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * External dependencies
+ */
+import { SocksProxyAgent } from 'socks-proxy-agent';
+import { HttpsProxyAgent } from 'https-proxy-agent';
+import { HttpProxyAgent } from 'http-proxy-agent';
+import { getProxyForUrl } from 'proxy-from-env';
+
```

Comment:
```suggestion
import debugLib from 'debug';
```
Adding some debug logs will help us debug when things don't go as expected.

Reply:
Added